MobiFlight/Joysticks/Joystick.cs

```diff
@@ -216,32 +246,33 @@ public List<ListItem<IBaseDevice>> GetAvailableDevicesAsListItems()
 
         private List<JoystickDevice> GetButtonsSorted()
         {
-            var buttons = Buttons.ToArray().ToList();
-            buttons.Sort(SortByPositionInDefintion);
-            return Buttons;
+            return Buttons.OrderBy(button => GetIndexForJoystickDevice(button)).ToList();
```

Comment:
This addresses two nuisances (cf. issue #1808):

- The mix of member "Buttons" and local "buttons" led to the sorted list being completely ignored
- List.Sort() uses an unstable sorting algorithm, effectively applying an undefined permutation to buttons without a sorting index.

The Lambda could also be a function pointer, writing it this way mainly makes it more explicit that it is acting on individual buttons.